docs/protocol.md

```diff
@@ -1062,6 +1063,118 @@ RPC
 ```
 </details>
 
+### Catching up on past messages on a channel using paging
+
+🧭 **RPC Message** > **Query** > **Paged Catchup**
+
+By executing a paged catchup action, a client can ask the server to receive a specified number of
+past messages on a specific channel.
+
+For now, it is to be used to retrieve chirps on the social media channel by paging when a new client joins the LAO instead of getting all chirps at once. This is done in an effort to reduce network traffic at catchup.
+
+This may serve as a starting point for the paging of messages in other channels as a future optimization.
+
+RPC 
+
+```json5
+// ../protocol/examples/query/paged_catchup/paged_catchup.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 3,
+    "method": "paged_catchup",
+    "params": {
+        "channel": "/root/p_EYbHyMv6sopI5QhEXBf40MO_eNoq7V_LygBd4c9RA=/social/chirps",
+        "number_of_messages": 10,
+        "before_message_id": "DCBX48EuNO6q-Sr42ONqsj7opKiNeXyRzrjqTbZ_aMI="
+    }
+}
+
+```
+
+Response (in case of success)
+
+
+```json5
+// ../protocol/examples/answer/general_message.json
+
+{
+    "jsonrpc": "2.0",
+    "id": 3,
+    "result": [
```

Comment:
would it make sense to return an object, containing this list but also the next page that should be retrieved ? I believe I've encouraged you to do so before, but you might want to look into pagination API designs. 
E.g. 
- https://nordicapis.com/everything-you-need-to-know-about-api-pagination/
- https://www.getknit.dev/blog/api-pagination-best-practices
- https://dev.to/pragativerma18/unlocking-the-power-of-api-pagination-best-practices-and-strategies-4b49
- etc.

I don't see in the current design a fully thought-through pagination system.
It looks as though you're doing a variant of keyset pagination, but you might want to think about all the use cases here. You can start with a simple one: assume you were up-to-date until yesterday, stayed offline since, and come back online now. How do you retrieve the latest messages with the current protocol ?

Reply:
"would it make sense to return an object, containing this list but also the next page that should be retrieved ?": 
If I want to build upon the current design, it would involve introducing page-based pagination on top of the keyset pagination variant currently used. One way I can think of to add the next page is by taking in the `number_of_messages` parameter and using that as the size of each page, but it might get messy if the user changes the value of this parameter. An alternative is to just remove this parameter and have fixed-size pages, which makes implementing paged-based pagination simpler but removes the flexibility that comes with that parameter.

"How do you retrieve the latest messages with the current protocol ?":
The client can request the latest messages one page at a time until they get a page containing a message they already have.

Reply:
> The client can request the latest messages one page at a time until they get a page containing a message they already have.
I think what @pierluca meant is that you should document how to retrieve the most recent set of messages using this protocol ;)